gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -587,7 +587,16 @@ def test_{{ method_name }}(request_type, transport: str = 'grpc'):
     {% for field in method.output.fields.values() | rejectattr('message') %}
     {% if not field.oneof or field.proto3_optional %}
     {% if field.field_pb.type in [1, 2] %}{# Use approx eq for floats #}
+    {% if field.repeated %}
+    for index in range(len(response.{{ field.name }})):
+        assert math.isclose(
+            response.{{ field.name }}[index],
+            {{ field.mock_value }}[index],
+            rel_tol=1e-6,
+        )
+    {% else %}{# field.repeated #}
```

Comment:
Is it feasible to refactor the same code?

Reply:
This is a much larger issue as duplication exists throughout the templates. We should tackle the duplicate code as a while separate from this bug fix. I've created https://github.com/googleapis/gapic-generator-python/issues/1592 to track the issue.